library/proc_macro/src/bridge/mod.rs

```diff
@@ -414,30 +466,7 @@ pub enum TokenTree<G, P, I, L> {
     Literal(L),
 }
 
-impl<G: Mark, P: Mark, I: Mark, L: Mark> Mark for TokenTree<G, P, I, L> {
```

Comment:
As someone new to this code, I'd recommend not doing this change. I understand that it's more general, and if there were lots of compound types needing marking/unmarking it would make more sense. But for a single type it's just more indirection and macros in a crate that is already stuffed to the gills with indirection and macros. Plus it makes the code longer.

Reply:
But there are quite a few types which use `compound_traits!` - by the end of the patch stack, it's not just `TokenTree`, but also `DelimSpan`, `Group`, `Punct`, `Ident`, `Literal`, and `ExpnContext`. Writing out marking/unmarking for each of them manually would be more painful than is warranted IMO.

I definitely agree that if we're considering the case of a single enum like  `TokenTree`, writing it out manually makes sense, but this isn't that case.

Reply:
Oh, sorry, I didn't notice it was used in the later commits.